ocs_ci/utility/kms.py

```diff
@@ -761,36 +764,304 @@ def create_vault_csi_kms_token(
-        )
-        if kv_version == "v1":
-            buf["VAULT_BACKEND"] = "kv"
+        if kms_type == constants.VAULT_TOKEN:
+            conn_str = csi_kms_conn_details["data"]["1-vault"]
+            buf = json.loads(conn_str)
+            buf["VAULT_ADDR"] = f"https://{self.vault_server}:{self.port}"
+            buf["VAULT_BACKEND_PATH"] = self.vault_backend_path
+            buf["VAULT_CACERT"] = get_default_if_keyval_empty(
+                config.ENV_DATA, "VAULT_CACERT", defaults.VAULT_DEFAULT_CA_CERT
+            )
+            buf["VAULT_NAMESPACE"] = self.vault_namespace
+            buf["VAULT_TOKEN_NAME"] = get_default_if_keyval_empty(
+                config.ENV_DATA,
+                "VAULT_TOKEN_NAME",
+                constants.EXTERNAL_VAULT_CSI_KMS_TOKEN,
+            )
+            if kv_version == "v1":
+                buf["VAULT_BACKEND"] = "kv"
+            else:
+                buf["VAULT_BACKEND"] = "kv-v2"
+
+            csi_kms_conn_details["data"]["1-vault"] = json.dumps(buf)
+
         else:
-            buf["VAULT_BACKEND"] = "kv-v2"
+            conn_str = csi_kms_conn_details["data"]["vault-tenant-sa"]
+            buf = json.loads(conn_str)
+            buf["vaultAddress"] = f"https://{self.vault_server}:{self.port}"
+            buf["vaultBackendPath"] = self.vault_backend_path
+            buf["vaultCAFromsecret"] = get_default_if_keyval_empty(
+                config.ENV_DATA, "VAULT_CACERT", defaults.VAULT_DEFAULT_CA_CERT
+            )
+            buf["vaultClientCertFromSecret"] = self.client_cert_name
+            buf["vaultClientCertKeyFromSecret"] = self.client_key_name
+            if self.vault_namespace:
+                buf["vaultNamespace"] = self.vault_namespace
+            if self.vault_kube_auth_path:
+                buf["vaultAuthPath"] = self.vault_kube_auth_path
+            else:
+                buf.pop("vaultAuthPath")
+            if self.vault_kube_auth_namespace:
+                buf["vaultAuthNamespace"] = self.vault_kube_auth_namespace
+            else:
+                buf.pop("vaultAuthNamespace")
+            csi_kms_conn_details["data"]["vault-tenant-sa"] = json.dumps(buf)
 
-        csi_kms_conn_details["data"]["1-vault"] = json.dumps(buf)
         csi_kms_conn_details["metadata"]["namespace"] = namespace
         self.create_resource(csi_kms_conn_details, prefix="csikmsconn")
 
+    def create_token_reviewer_resources(self):
+        """
+        This function will create the rbd-csi-vault-token-review SA, clusterRole and
+        clusterRoleBindings required for the kubernetes auth method with vaulttenantsa
+        encryption type.
+
+        """
+
+        try:
+            rbd_vault_token_reviewer = templating.load_yaml(
+                constants.RBD_CSI_VAULT_TOKEN_REVIEWER, multi_document=True
+            )
+            self.create_resource(rbd_vault_token_reviewer, prefix="rbd-token-review")
```

Comment:
please split the params in seperate lines.
```suggestion
            self.create_resource(
                rbd_vault_token_reviewer, 
                prefix="rbd-token-review"
             )
```

Reply:
Done

Reply:
ack